REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/certificate/hateoas/CertificateHateoasSupport.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.esm.certificate.hateoas;
+
+import com.epam.esm.certificate.controller.CertificateController;
+import com.epam.esm.certificate.entity.Certificate;
+import com.epam.esm.tag.hateoas.TagHateoasSupport;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
+
+import java.util.List;
+
+@Configuration
+public class CertificateHateoasSupport {
```

Comment:
Why Configuration?

Reply:
parameter  of constructor in each controller required a bean for these classes so i have anotate them as `@Configuration` or `@Component`

Reply:
Its probably wrong to use an Configuration annotation here and better to replace it with Component